sycl/doc/extensions/experimental/sycl_ext_oneapi_properties.asciidoc

```diff
@@ -169,76 +170,49 @@ namespace sycl::ext::oneapi::experimental {
 template<typename...> struct property_value;
 
 // This property has no parameters.
-struct bar_key {
-  using value_t = property_value<bar_key>;
-};
-// bar is an object of the property value type of bar.
-inline constexpr bar_key::value_t bar;
+inline constexpr property_value<struct bar_key> bar;
```

Comment:
Should we allow this? Or should we require that keys are not only declared but also defined:
```
struct bar_key {};
inline constexpr property_value<bar_key> bar;
```
See also https://github.com/intel/llvm/pull/13776#pullrequestreview-2074340326

Reply:
At this point, the property key class is only used as the template parameter to `has_property`, `get_property`, and `is_property_key_of`, correct?  Is there ever a need to instantiate an object of the key class?  If not, then it seems OK to allow it to be an incomplete type.

I didn't understand the concern in #13776.  Is there a reason we should not allow the type to be incomplete?

Reply:
If a runtime property (=non-empty) were incomplete `is_empty_or_incomplete` (`IsRuntimeProperty` or `get_property`) the implementation would falsely assume it is a compile-time property. This could happen e.g. if the property isn't declared by the include and the user fwd-declares it. I think even then then the worst that could happen is a bit misleading error message.

Reply:
If we required compile-time property keys to be a complete type (i.e. empty struct), you are saying that we could give a better error message in the scenario you describe?

I think there is not a big downside if we require compile-time property keys to be an empty struct.  So, if that gives us better error messages, it seems reasonable.

Reply:
@rolandschulz: It seems like this issue is not resolved, and I think our recent discussions are moving to the direction where each property will have a "_key" that is a complete type.  Should we revert this part of the change?  Note that some of the other changes below also remove the "_key" struct, replacing it with an incomplete type.

Reply:
Resolved offline.  We think it is OK for the "_key" type to be incomplete.